pandas/core/arrays/datetimelike.py

```diff
@@ -309,14 +309,28 @@ def __array__(self, dtype: NpDtype | None = None) -> np.ndarray:
             return np.array(list(self), dtype=object)
         return self._ndarray
 
+    @overload
+    def __getitem__(self, item: int | np.integer) -> DTScalarOrNaT:
+        ...
+
+    @overload
+    def __getitem__(
+        self, item: slice | np.ndarray | Sequence[int]
+    ) -> DatetimeLikeArrayMixin:
+        ...
+
     def __getitem__(
         self, key: PositionalIndexer2D
     ) -> DatetimeLikeArrayMixin | DTScalarOrNaT:
         """
         This getitem defers to the underlying array, which by-definition can
         only handle list-likes, slices, and integer scalars
         """
-        result = super().__getitem__(key)
+        #  error: Invalid index type "Union[Union[int, integer[Any], slice,
+        # Sequence[int], ndarray], Tuple[Union[int, integer[Any], slice, Sequence[int],
+        # ndarray], Union[int, integer[Any], slice, Sequence[int], ndarray]]]" for
+        # "DatetimeLikeArrayMixin"; expected type "Union[int, integer[Any]]"
+        result = super().__getitem__(key)  # type: ignore[index]
```

Comment:
best guess is we need _libs/arrays.pyi

Reply:
> best guess is we need _libs/arrays.pyi

But `_libs/arrays.pyx` doesn't define `__getitem__`.  What I can't figure out is why it thinks it is expecting `Union[int, integer[Any]]` when the `super()` should be referring to `NDArrayBackedExtensionArray` , which is a subclass of `ExtensionArray`.  Even if I use `super(NDArrayBackedExtensionArray, self)` I get the same mypy message.

Reply:
Fixed with a `cast`